inspire/dojson/utils.py

```diff
@@ -85,3 +85,23 @@ def create_profile_url(profile_id):
         return base_url + str(profile_id)
     except (TypeError, ValueError):
         return ''
+
+
+def full_date_format(date):
```

Comment:
I would recommend a different way of doing the date parsing by using existing libraries. E.g. `datetime` and `dateutil`.

```python
from dateutil.parser import parse
import datetime

# Need default to get proper default values if parts are missing
parse("2015-01-22", default=datetime.datetime(year=1900, day=1, month=1))
# Out[10]: datetime.datetime(2015, 1, 22, 0, 0)

parse("2015-12", default=datetime.datetime(year=1900, day=1, month=1))
# Out[11]: datetime.datetime(2015, 12, 1, 0, 0)

parse("2015", default=datetime.datetime(year=1900, day=1, month=1))
# Out[12]: datetime.datetime(2015, 1, 1, 0, 0)

# Then use strftime to print the correct format
parse("2015", default=datetime.datetime(year=1900, day=1, month=1)).strftime("%Y-%m-%d")
# Out[13]: '2015-01-01'
```